src/shared/vscode/env.ts

```diff
@@ -0,0 +1,23 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+// @ts-ignore
+import * as vscode from 'vscode'
+
+/**
+ * Components associated with {@link vscode.env}.
+ */
+export interface Env {
```

Comment:
Do you envision objects like this getting passed around to every function in many call chains? Curious to understand the usage.

Reply:
I expect it to be injected into the class/function that needs it. So the caller shouldn't need to pass it around.

I've been using it as a default parameter on (command) class constructors. [(example)](https://github.com/aws/aws-toolkit-vscode/blob/kantc/4commands/src/awsexplorer/commands/copyName.ts#L17).

```ts
export async function copyNameCommand(
    node: AWSResourceNode,
    window: Window = new DefaultWindow(),
    env: Env = new DefaultEnv()
): Promise<void> {
```

The Window and Env fakes would be injected for unit testing. This doesn't solve the problem of trying to test a class which uses a class which uses this. We'd need real dependency injection or module replacement to do that. 

However, I've found that unit tests are quite sufficient for testing the kind of edge cases and strict validation that this is intended to help verify. Integration tests can usually use the real implementations and perform less precise validations.